src/main/java/us/abstracta/wiresham/FlowFactory.java

```diff
@@ -0,0 +1,187 @@
+import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.constructor.Constructor;
+import org.yaml.snakeyaml.introspector.Property;
+import org.yaml.snakeyaml.nodes.NodeTuple;
+import org.yaml.snakeyaml.nodes.Tag;
+import org.yaml.snakeyaml.representer.Representer;
+
+public class FlowFactory {
+
+  private static final Logger LOG = LoggerFactory.getLogger(FlowFactory.class);
+  private static final Map<String, Class<?>> YAML_TAGS = ImmutableMap.<String, Class<?>>builder()
+      .put("!server", SendPacketStep.class)
+      .put("!client", ReceivePacketStep.class)
+      .build();
+
+  private final File configFile;
+  private final String serverAddress;
+  private final String pcapFilter;
+  private final int port;
+  private final String hostTarget;
+
+  public FlowFactory(File configFile, int port, String serverAddress, String pcapFilter,
+      String targetAddress) {
+    this.configFile = configFile;
+    this.serverAddress = serverAddress;
+    this.pcapFilter = pcapFilter;
+    this.port = targetAddress == null ? port : getPortFromTargetAddress(targetAddress);
+    this.hostTarget = getHostFromTargetAddress(targetAddress);
+  }
+
+  private String getHostFromTargetAddress(String targetAddress) {
+    if (targetAddress == null) {
+      return null;
+    }
+    int hostPortSeparator = targetAddress.lastIndexOf(":");
+    return targetAddress.substring(0, hostPortSeparator);
+  }
+
+  private int getPortFromTargetAddress(String targetAddress) {
+    int portSeparatorPos = targetAddress.lastIndexOf(":");
+    return Integer.parseInt(targetAddress.substring(portSeparatorPos + 1));
+  }
+
+  public List<Flow> getFlows() {
+    List<Flow> flows;
+    try {
+      if (serverAddress != null) {
+        flows = Collections.singletonList(buildWiresharkFlow());
+      } else {
+        flows = buildFromYamlFile();
+      }
+
+      if (flows.size() == 1 && flows.get(0).getPort() == null) {
+        flows.get(0).setPort(port);
+      }
+      return flows;
+    } catch (IOException e) {
+      LOG.error("There was an error trying to read the wireshark flow", e);
+      System.err.println(e.getMessage());
+      return null;
+    }
+  }
+
+  private List<Flow> buildFromYamlFile() throws FileNotFoundException {
+    List<PacketStep> packets = new Yaml(buildYamlConstructor())
+        .load(new FileInputStream(configFile));
+    return dissectStepsIntoFlow(packets);
+  }
+
+  private static Constructor buildYamlConstructor() {
+    Constructor constructor = new Constructor();
+    YAML_TAGS
+        .forEach((tag, clazz) -> constructor.addTypeDescription(new TypeDescription(clazz, tag)));
+    return constructor;
+  }
+
+  public List<Flow> dissectStepsIntoFlow(List<PacketStep> packetSteps) {
+    List<List<PacketStep>> dissectedFlow = new ArrayList<>();
+    List<PacketStep> currentFlowPort = new ArrayList<>();
+    for (PacketStep step : packetSteps) {
+      if ((step instanceof SendPacketStep) && ((SendPacketStep) step).getPort() != null) {
```

Comment:
what happens when you have two send packets with port? Does it separate them in two flows?. Eg:

```yaml
- !server: {port: 2324, data: xxx}
- !server: {port: 2323, data: yyy}
```